src/util/config.cpp

```diff
@@ -8,10 +8,20 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include <algorithm>
 #include <regex>
+#include <thread>
 
 #include <util/config.h>
 
-configt config;
+configt *configt::config = nullptr;
+std::mutex configt::instance_mutex;
+
+configt *configt::get_instance()
+{
+  std::lock_guard<std::mutex> lock(instance_mutex);
```

Comment:
Don't we have to call `unlock` somewhere?

Reply:
No need, it is released automatically when the lock goes out of scope.